src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_transformers.py

```diff
@@ -61,3 +61,47 @@ def pricesheet_show_properties(result):
 def transform_pricesheet_show_output(result):
     result.pricesheets = [pricesheet_show_properties(item) for item in result.pricesheets]
     return result
+
+
+def marketplace_list_output(result):
+    result.pretax_cost = str(result.pretax_cost)
+    result.consumed_quantity = str(result.consumed_quantity)
+    result.resource_rate = str(result.resource_rate)
+    return result
+
+
+def budget_list_output(result):
+    result.amount = str(result.amount)
+    if result.current_spend:
+        result.current_spend.amount = str(result.current_spend.amount)
```

Comment:
I still wonder why this is needed only for consumption.  Other services don't have to do this. Does this have some kind of precision impact that might be an issue?  For example, does it simply truncate the value or does it round?  @lmazuel is there a swagger tweak that would make this unnecessary?

Reply:
Budget model sets amount and current spend as decimal. So we convert them to string to be compatible with Python